tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+        with self.assertRaises((pexEx.OutOfRangeError, OverflowError)):
+            foo = point[-1]
+
+        foo = point[0]
+        foo = point[1]
+
+    def testGetItemValue(self):
+        """Test if indexing returns the expected value.
+        """
+        for lon, lat in self._dataset:
+            point = SphPoint(lon*degrees, lat*degrees)
+            self.assertIsInstance(point[0], afwGeom.Angle)
+            self.assertIsInstance(point[1], afwGeom.Angle)
+            self.assertEqual(point.getLongitude(), point[0])
+            self.assertEqual(point.getLatitude(),  point[1])
+
+    def testEquality(self):
+        """Test if tests for equality treat SphPoints as values.
+        """
+        # (In)equality is determined by value, not identity.
+        # See DM-2347, DM-2465. These asserts are testing the
+        # functionality of `==` and `!=` and should not be changed.
+        lon, lat = self._dataset[0]
+        point1 = SphPoint(lon*degrees, lat*degrees)
+        self.assertIsInstance(point1 == point1, bool)
+        self.assertIsInstance(point1 != point1, bool)
+        self.assertTrue(point1 == point1)
+        self.assertFalse(point1 != point1)
+
+        lon, lat = self._dataset[1]
+        point2 = SphPoint(lon*degrees, lat*degrees)
+        self.assertTrue(point2 != point1)
+        self.assertTrue(point1 != point2)
+        self.assertFalse(point2 == point1)
+        self.assertFalse(point1 == point2)
+
+        pointCopy = copy.deepcopy(point1)
+        self.assertEqual(pointCopy, point1)
+        self.assertEqual(point1, pointCopy)
+        self.assertFalse(pointCopy != point1)
+        self.assertFalse(point1 != pointCopy)
+
+        # Test for transitivity (may be assumed by algorithms).
+        for delta in [10.0**(0.1*x) for x in range(-150, -49, 5)]:
+            self._checkTransitive(delta)
+
+    def _checkTransitive(self, delta):
+        """Test if equality is transitive even for close points.
+
+        This test prevents misuse of approximate floating-point
+        equality -- if `__eq__` is implemented using AFP, then this
+        test will fail for some value of `delta`. Testing multiple
+        values is recommended.
+
+        Parameters
+        ----------
+        delta : `number`
+            The separation, in degrees, at which point equality may
+            become intransitive.
+        """
+        point1 = SphPoint((45.0 - delta)*degrees, -18.0*degrees);
+        point2 = SphPoint( 45.0         *degrees, -18.0*degrees);
+        point3 = SphPoint((45.0 + delta)*degrees, -18.0*degrees);
+
+        self.assertTrue(point1 != point2 or point2 != point3 or point1 == point3)
+        self.assertTrue(point3 != point1 or point1 != point2 or point3 == point2)
+        self.assertTrue(point2 == point3 or point3 != point1 or point2 == point1)
+
+    def testEqualityAlias(self):
+        """Test if == handles coordinate degeneracies correctly.
+        """
+        # Coordinate wrapping
```

Comment:
I suggest `# Longitude wrapping`. Let's save "coord" for the `Coord` class.

Speaking of which, you use the variable "coord" elsewhere and should consider renaming it, e.g. to longLat